python/ray/data/dataset.py

```diff
@@ -379,7 +379,7 @@ def map_batches(
         *,
         batch_size: Optional[Union[int, Literal["default"]]] = "default",
         compute: Optional[Union[str, ComputeStrategy]] = None,
-        batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+        batch_format: Optional[str] = "default",
```

Comment:
keep it as `Optional[Literal]` for full explicitness of supported batch formats?

Reply:
I feel like that is hard to maintain (per the inconsistencies in the code already), so opted to go unify on the shorter signature.